src/prototype_creep_startup_tasks.js

```diff
@@ -463,9 +463,6 @@ Creep.prototype.repairStructure = function() {
 };
 
 Creep.prototype.getDroppedEnergy = function() {
-  if (this.carry.energy > 0.9 * this.carryCapacity) {
-    return false;
-  }
```

Comment:
I don't think this is necessary given the state machine, but I can keep it if you want.